exo/topology/device_capabilities.py

```diff
@@ -77,24 +76,60 @@ def device_capabilities() -> DeviceCapabilities:
     else:
         return DeviceCapabilities(model=f"Unknown Device", chip=f"Unknown Chip", memory=psutil.virtual_memory().total // 2**20, flops=DeviceFlops(fp32=0, fp16=0, int8=0))
 
+
+import subprocess
+import re
+
+
 def mac_device_capabilities() -> DeviceCapabilities:
     # Fetch the model of the Mac using system_profiler
-    model = subprocess.check_output(['system_profiler', 'SPHardwareDataType']).decode('utf-8')
-    model_line = next((line for line in model.split('\n') if "Model Name" in line), None)
-    model_id = model_line.split(': ')[1] if model_line else "Unknown Model"
-    chip_line = next((line for line in model.split('\n') if "Chip" in line), None)
-    chip_id = chip_line.split(': ')[1] if chip_line else "Unknown Chip"
-    memory_line = next((line for line in model.split('\n') if "Memory" in line), None)
-    memory_str = memory_line.split(': ')[1] if memory_line else "Unknown Memory"
-    memory_units = memory_str.split()
-    memory_value = int(memory_units[0])
-    if memory_units[1] == "GB":
-        memory = memory_value * 1024
-    else:
-        memory = memory_value
-
-    # Assuming static values for other attributes for demonstration
-    return DeviceCapabilities(model=model_id, chip=chip_id, memory=memory, flops=CHIP_FLOPS.get(chip_id, DeviceFlops(fp32=0, fp16=0, int8=0)))
+    hw_info = subprocess.check_output(['system_profiler', 'SPHardwareDataType']).decode('utf-8')
```

Comment:
I just took another look and there's a `-json` flag that might make parsing a bit cleaner: `system_profiler SPHardwareDataType -json`

Reply:
Indeed, removed "re" and used json processing instead: 

```json
system_profiler SPHardwareDataType -json
{
  "SPHardwareDataType" : [
    {
      "_name" : "hardware_overview",
      "activation_lock_status" : "activation_lock_disabled",
      "boot_rom_version" : "10151.101.3",
      "chip_type" : "Unknown",
      "machine_model" : "MacBookPro18,2",
      "machine_name" : "MacBook Pro",
      "model_number" : "Z14X000PRFN/A",
      "number_processors" : "proc 10:8:2",
      "os_loader_version" : "10151.101.3",
      "physical_memory" : "64 GB",
      "platform_UUID" : "AE593C1A-1051-5AE0-AA12-69CE572FB403",
      "provisioning_UDID" : "AE593C1A-1051-5AE0-AA12-69CE572FB403",
      "serial_number" : "WR295HC54D"
    }
  ]
}
```